sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/patterns.scala

```diff
@@ -159,18 +170,16 @@ object PartialAggregation {
             // Should trim aliases around `GetField`s. These aliases are introduced while
             // resolving struct field accesses, because `GetField` is not a `NamedExpression`.
             // (Should we just turn `GetField` into a `NamedExpression`?)
-            namedGroupingExpressions
+            namedGroupingMap
               .get(e.transform { case Alias(g: ExtractValue, _) => g })
               .map(_.toAttribute)
               .getOrElse(e)
         }).asInstanceOf[Seq[NamedExpression]]
 
         val partialComputation =
-          (namedGroupingExpressions.values ++
+          (partialGroupingExprs ++
             partialEvaluations.values.flatMap(_.partialEvaluations)).toSeq
 
-        val namedGroupingAttributes = namedGroupingExpressions.values.map(_.toAttribute).toSeq
```

Comment:
`namedGroupingExpressions.values` probably come with arbitrary order, which is not right compare to the `groupingExpressions`.